src/main/java/org/tinder/servlets/MessageServlet.java

```diff
@@ -0,0 +1,85 @@
+package org.tinder.servlets;
+
+import freemarker.template.TemplateException;
+import org.tinder.enums.CookieNames;
+import org.tinder.models.Message;
+import org.tinder.models.User;
+import org.tinder.services.MessageService;
+import org.tinder.services.UserService;
+import org.tinder.utils.CookieWorker;
+import org.tinder.utils.FMTemplate;
+
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.List;
+
+public class MessageServlet extends HttpServlet{
+
+    private final MessageService messageService;
+
+    public MessageServlet(){
+
+            messageService = new MessageService();
+
+    }
+
+    // http://localhost:8080/messages
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+
+        Integer id = getMessageId(req);
+        String token = CookieWorker.getCookieOrThrow(req, CookieNames.AUTH_TOKEN);
+        User user = new UserService().login(token);
```

Comment:
we already has user service in services. 

you can use MessageServlet extends ServicesServlet
and super services

services prop ->  new MessageServlet(services) in TinderApp

Reply:
+